# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLVertexArray.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLVertexArray.cpp
--- a/dom/canvas/WebGLVertexArray.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLVertexArray.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -14,14 +14,9 @@
 
 namespace mozilla {
 
-WebGLVertexArray::WebGLVertexArray(WebGLContext* const webgl, const GLuint name)
-    : WebGLRefCountedObject(webgl),
-      mGLName(name),
-      mAttribs(mContext->mGLMaxVertexAttribs) {
-  mContext->mVertexArrays.insertBack(this);
-}
-
-WebGLVertexArray::~WebGLVertexArray() { MOZ_ASSERT(IsDeleted()); }
+WebGLVertexArray::WebGLVertexArray(WebGLContext* const webgl)
+    : WebGLContextBoundObject(webgl),
+      mAttribs(mContext->Limits().maxVertexAttribs) {}
 
 WebGLVertexArray* WebGLVertexArray::Create(WebGLContext* webgl) {
   WebGLVertexArray* array;
@@ -33,30 +28,4 @@
   return array;
 }
 
-void WebGLVertexArray::Delete() {
-  DeleteImpl();
-
-  LinkedListElement<WebGLVertexArray>::removeFrom(mContext->mVertexArrays);
-  mElementArrayBuffer = nullptr;
-  mAttribs.clear();
-}
-
-// -
-
-inline void ImplCycleCollectionTraverse(
-    nsCycleCollectionTraversalCallback& callback,
-    const std::vector<WebGLVertexAttribData>& field, const char* name,
-    uint32_t flags = 0) {
-  for (auto& cur : field) {
-    ImplCycleCollectionTraverse(callback, cur.mBuf, name, flags);
-  }
-}
-
-inline void ImplCycleCollectionUnlink(
-    std::vector<WebGLVertexAttribData>& field) {
-  for (auto& cur : field) {
-    cur.mBuf = nullptr;
-  }
-}
-
 }  // namespace mozilla